src/FavIcon.vala

```diff
@@ -37,14 +37,24 @@ public class FeedReader.FavIcon : GLib.Object
 	private Gee.Promise<Gdk.Pixbuf?> m_pixbuf = null;
 	private ResourceMetadata m_metadata;
 
-	public signal void pixbuf_changed(Feed feed, Gdk.Pixbuf pixbuf);
+	public signal void surface_changed(Feed feed, Cairo.Surface surface);
 
 	private FavIcon(Feed feed)
 	{
 		m_feed = feed;
 	}
 
-	public async Gdk.Pixbuf? get_pixbuf()
+	private int GetScaleFactor()
+	{
+		return MainWindow.get_default().get_style_context().get_scale();
+	}
+
+	private Cairo.Surface CreateSurfaceFromPixbuf(Gdk.Pixbuf pixbuf)
+	{
+		return Gdk.cairo_surface_create_from_pixbuf(pixbuf, GetScaleFactor(), null);
+	}
```

Comment:
Since these two functions are both only used in `get_surface()` it seems simpler to just put this in `get_surface()`: 
```
var scale = MainWindow.get_default().get_style_context().get_scale();
return Gdk.cairo_surface_create_from_pixbuf(pixbuf, scale, null);
```

Reply:
We also need the scale in L96, hence the function.

Reply:
But the two uses are right next to each other in the same function. A variable seems to make more sense than a function?

Reply:
makes sense

Reply:
Oh wait, we do call `CreateSurfaceFromPixbuf` twice.. so nevermind, your version makes sense :D